### PR TITLE
[ecm] update version to 6.4.0 and fix absolute paths

### LINKED ISSUE #40244: [ecm] update to >=6.0.0
### Library name

ecm

### New version number

\>=6.0.0

### Other information that may be useful (release notes, etc...)

_No response_

### PR DESCRIPTION
**Pull request by JonLiu1993**
Fixes #40244

1. Update port version to 6.4.0
2. Fix absolute paths
````
F:\test\vcpkg\ports\ecm\portfile.cmake: warning: There should be no absolute paths, such as the following, in an installed package. To suppress this message, add set(VCPKG_POLICY_SKIP_ABSOLUTE_PATHS_CHECK enabled)
note: F:\test\vcpkg\packages
note: F:\test\vcpkg\installed
note: F:\test\vcpkg\buildtrees
note: F:\test\vcpkg\downloads
note: Absolute paths were found in the following files
F:\test\vcpkg\packages\ecm_x64-windows: note: the files are relative to ${CURRENT_PACKAGES_DIR} here
note: share/ECM/cmake/ECMConfig.cmake
F:\test\vcpkg\ports\ecm\portfile.cmake: warning: Found 1 post-build check problem(s). These are usually caused by bugs in portfile.cmake or the upstream build system. Please correct these before submitting this port to the curated registry.
````

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version.~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

No feature need to test.
tested usage successfully by `ecm:x64-windows`:
````
The package ecm provides CMake targets:

    find_package(ECM CONFIG REQUIRED NO_MODULE)
    set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
    include(KDEInstallDirs)
    include(KDECompilerSettings)
    include(KDECMakeSettings)
````

### CODE REVIEW

**Comment by dg0yt** [comment]
~~~diff
-The package ecm provides CMake targets:
+ecm provides CMake files:
~~~

**Comment by dg0yt** [comment]
~~~diff
-    set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH})
+    list(INSERT CMAKE_MODULE_PATH 0 ${ECM_MODULE_PATH})
~~~

**Review by jimwang118 — APPROVED** [review]

File: ports/ecm/portfile.cmake [hunk 1]
```diff
@@ -1,12 +1,10 @@
...
     OUT_SOURCE_PATH SOURCE_PATH
     REPO KDE/extra-cmake-modules
-    REF v5.98.0
-    SHA512 355bf2010dd4c736981d0f708ab7245c0925104e19da01acee0482494f815088e5d7aed9aa968251f13c2c5b8cbfd67dc8bb147dec46050fea0881c00f456d9d
+    REF "v${VERSION}"
+    SHA512 30f9e4d606f88c8d0e2be8f0fbba47ae25cf6e1c9716d50fc2f7845e5527d6cb4a34dfca8ec4356e8e74ce936ae252edf833b2bc545a4c85a894103dd27d8c89
     HEAD_REF master
     PATCHES
-        fix_canberra.patch         # https://invent.kde.org/frameworks/extra-cmake-modules/-/merge_requests/187
-        fix_libmount.patch         # https://invent.kde.org/frameworks/extra-cmake-modules/-/merge_requests/200
```
Comment (JonLiu1993) [inline ports/ecm/portfile.cmake:9]: These two versions were already applied in the latest version, so I deleted them.

**Review by data-queue — APPROVED** [review]

**Pull request merged**